### PR TITLE
Added pydocstyle to pre-commit and tox

### LINKED ISSUE #4223: Add pydocstyle check onto pre-commit
As the documentation mentions, we are using pydocstyle the docstrings. There is an available hook for this to use.

But we might be better off waiting for #4131 to be resolved whenever I got a spare hour on doing that.

The hook documentation is available here https://github.com/jans-forks/pydocstyle/blob/1fa421f76a76bd2460aaf2e7e025ee00e9ec70e6/docs/snippets/pre_commit.rst.

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
Closes #4223.

Disables unneeded checks:
- D407: required dashed line under sections, e.g. `Args: ....`;
- D202: no blank line after a docstring (conflict with black, and having a blank line makes it easier to read).

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 03e83b2390e7fc8f032f5bef56beaaf9c590bc85 (NyanKiyoshi/saleor @ pre-commit/add-pydocstyle)
Base comparison is d9a4ecbb651e191c84443cba81a49924a123716b.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         41	         41	             20

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

File: saleor/extensions/plugins/avatax/__init__.py [hunk 1]
```diff
@@ -150,8 +151,11 @@ def checkout_needs_new_fetch(data, checkout_token: str) -> bool:
 
 
 def taxes_need_new_fetch(data: Dict[str, Any], taxes_token: str) -> bool:
-    """We store the response from avatax. If object doesn't exist in cache or
-    something has changed, then we fetch data from avatax."""
+    """We store the response from avatax.
```
Comment (maarcingebala) [inline saleor/extensions/plugins/avatax/__init__.py]: I know it was here before but now it looks even more strange with this formatting applied. We could use a docstring like this:

```
"""Check if Avatax's taxes data need to be refetched.

The response from Avatax is stored in a cache. If an object doesn't exist in cache or something has changed, taxes need to be refetched.
"""
```

The docstring for `checkout_needs_new_fetch` should follow the same convention.

Comment (maarcingebala) [reply saleor/extensions/plugins/avatax/__init__.py]: @korycins Please review this part as well, as you're the author of the original docstrings (I guess).

File: saleor/graphql/core/enums.py [hunk 2]
```diff
@@ -15,8 +15,7 @@ class ReportingPeriod(graphene.Enum):
 
 
 def to_enum(enum_cls, *, type_name=None, **options) -> graphene.Enum:
-    """
-    Create a graphene enum from a class containing a set of options.
+    """Create a graphene enum from a class containing a set of options.
```
Comment (maarcingebala) [inline saleor/graphql/core/enums.py]:
```suggestion
    """Create a Graphene enum from a class containing a set of options.
```

File: saleor/graphql/core/filters.py [hunk 3]
```diff
@@ -26,8 +26,10 @@ def validate(self, value):
 
 
 class EnumFilter(django_filters.CharFilter):
-    """ Filter class for graphene enum object.
-    enum_class needs to be passed explicitly  as well as the method."""
+    """Filter class for graphene enum object.
```
Comment (maarcingebala) [inline saleor/graphql/core/filters.py]:
```suggestion
    """Filter class for Graphene enum object.
```

File: saleor/graphql/core/types/filter_input.py [hunk 4]
```diff
@@ -35,9 +37,10 @@ def __init_subclass_with_meta__(
 
     @classmethod
     def get_filtering_args_from_filterset(cls):
-        """ Inspect a FilterSet and produce the arguments to pass to
-            a Graphene Field. These arguments will be available to
-            filter against in the GraphQL
+        """Retrieve the filtering arguments from the queryset.
+
+        Inspect a FilterSet and produce the arguments to pass to a Graphene Field.
```
Comment (maarcingebala) [inline saleor/graphql/core/types/filter_input.py]:
```suggestion
        Inspect a FilterSet and produce the arguments to pass to a Graphene field.
```

File: saleor/graphql/middleware.py [hunk 5]
```diff
@@ -5,8 +5,7 @@
 
 
 def jwt_middleware(get_response):
-    """Authenticate user using JSONWebTokenMiddleware
-    ignoring the session-based authentication.
+    """Authenticate user using JWT and ignoring the session-based authentication.
```
Comment (maarcingebala) [inline saleor/graphql/middleware.py]:
```suggestion
    """Authenticate a user using JWT and ignore the session-based authentication.
```

File: saleor/order/models.py [hunk 6]
```diff
@@ -375,8 +375,7 @@ def quantity_unfulfilled(self):
 
     @property
     def is_digital(self) -> bool:
-        """Return true if product variant is a digital type and has assigned
-        digital content"""
+        """Whether the variant is digital and does contain digital content."""
```
Comment (maarcingebala) [inline saleor/order/models.py]: Not exactly sure about this but I I think this should be correct:
```suggestion
        """Check if a variant is digital and contains digital content."""
```

File: saleor/payment/gateways/razorpay/__init__.py [hunk 7]
```diff
@@ -45,14 +44,16 @@ def _generate_response(
...
-    """Checks that a given payment is supported."""
+    """Check that a given payment is supported."""
     if payment_information.currency not in SUPPORTED_CURRENCIES:
         return errors.UNSUPPORTED_CURRENCY % {"currency": payment_information.currency}
 
 
 def get_error_message_from_razorpay_error(exc: BaseException):
-    """Convert a error razorpay error to a user friendly error message
-    and log the exception to stderr."""
+    """Convert a error razorpay error to a user friendly error message.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/__init__.py]:
```suggestion
    """Convert a Razorpay error to a user-friendly error message.
```

File: saleor/payment/gateways/razorpay/__init__.py [hunk 8]
```diff
@@ -45,14 +44,16 @@ def _generate_response(
...
+
+    It also log the exception to stderr.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/__init__.py]:
```suggestion
    It also logs the exception to stderr.
```

File: saleor/payment/gateways/razorpay/__init__.py [hunk 9]
```diff
@@ -61,8 +62,11 @@ def get_error_message_from_razorpay_error(exc: BaseException):
 
 
 def clean_razorpay_response(response: Dict):
-    """As the Razorpay response payload contains the final amount
-    in Indian rupees, we convert the amount to paisa (by dividing by 100)."""
+    """Convert the razorpay response to our internal represation for easier processing.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/__init__.py]:
```suggestion
    """Convert the Razorpay response to our internal representation for easier processing.
```

File: saleor/payment/gateways/razorpay/__init__.py [hunk 10]
```diff
@@ -88,13 +92,13 @@ def get_client_token(**_):
...
     """Capture a authorized payment using the razorpay client.
+
     But it first check if the given payment instance is supported
     by the gateway.
 
-    If an error from razorpay occurs,
-    we flag the transaction as failed and return
-    a short user friendly description of the error
-    after logging the error to stderr."""
+    If an error from razorpay occurs, we flag the transaction as failed and return
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/__init__.py]:
```suggestion
    If an error from Razorpay occurs, we flag the transaction as failed and return
```

File: saleor/payment/gateways/razorpay/__init__.py [hunk 11]
```diff
@@ -131,7 +135,8 @@ def refund(payment_information: PaymentData, config: GatewayConfig) -> GatewayRe
     It first retrieve a `charge` transaction to retrieve the
     payment id to refund. And return an error with a failed transaction
     if the there is no such transaction, or if an error
-    from razorpay occurs during the refund."""
+    from razorpay occurs during the refund.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/__init__.py]:
```suggestion
    from Razorpay occurs during the refund.
```

File: saleor/payment/gateways/razorpay/utils.py [hunk 12]
```diff
@@ -2,12 +2,16 @@
 
 
 def get_error_response(amount: Decimal, **additional_kwargs) -> dict:
-    """Create a place holder response for invalid/ failed requests
-    for generated a failed transaction object."""
+    """Create a place holder response for invalid or failed requests.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/utils.py]:
```suggestion
    """Create a placeholder response for invalid or failed requests.
```

File: saleor/payment/gateways/razorpay/utils.py [hunk 13]
```diff
@@ -2,12 +2,16 @@
...
+
+    It is used to generate a failed transaction object.
+    """
     return {"is_success": False, "amount": amount, **additional_kwargs}
 
 
 def get_amount_for_razorpay(amount: Decimal) -> int:
-    """Convert an amount of Indian rupees to paisa (needed by the razorpay)
-    by multiplying the value by 100."""
+    """Convert an amount of Indian rupees to paisa (needed by the razorpay).
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/utils.py]:
```suggestion
    """Convert an amount of Indian rupees to paisa (needed by the Razorpay).
```

File: saleor/payment/gateways/razorpay/utils.py [hunk 14]
```diff
@@ -2,12 +2,16 @@
...
+
+    It simply multiply the value by 100.
```
Comment (maarcingebala) [inline saleor/payment/gateways/razorpay/utils.py]:
```suggestion
    Multiplies the value by 100.
```

File: saleor/payment/gateways/stripe/__init__.py [hunk 15]
```diff
@@ -16,8 +16,10 @@
 
 
 def get_client_token(**_):
-    """Not implemented for stripe gateway currently. The client token can be
-    generated by Stripe's checkout.js or stripe.js automatically.
+    """Not implemented for stripe gateway currently.
```
Comment (maarcingebala) [inline saleor/payment/gateways/stripe/__init__.py]:
```suggestion
    """Not implemented for Stripe gateway currently.
```

File: saleor/payment/gateways/stripe/forms.py [hunk 16]
```diff
@@ -60,7 +60,8 @@ def render(self, name=None, value=None, attrs=None, renderer=None):
 
 
 class StripePaymentModalForm(forms.Form):
-    """
+    """The modal form of the stripe payment provider.
```
Comment (maarcingebala) [inline saleor/payment/gateways/stripe/forms.py]:
```suggestion
    """The modal form of the Stripe payment provider.
```

File: saleor/payment/utils.py [hunk 17]
```diff
@@ -280,7 +281,7 @@ def clean_mark_order_as_paid(order: Order):
 
 
 def call_gateway(operation_type, payment, payment_token, **extra_params):
-    """Helper that calls the passed gateway function and handles exceptions.
+    """Call the passed gateway function and handles exceptions.
```
Comment (maarcingebala) [inline saleor/payment/utils.py]:
```suggestion
    """Call the passed gateway function and handle exceptions.
```

File: saleor/payment/utils.py [hunk 18]
```diff
@@ -419,10 +419,12 @@ def gateway_process_payment(
 
 @require_active_payment
 def gateway_authorize(payment: Payment, payment_token: str) -> Transaction:
-    """Authorizes the payment and creates relevant transaction.
+    """Authorize the payment and creates relevant transaction.
```
Comment (maarcingebala) [inline saleor/payment/utils.py]:
```suggestion
    """Authorize the payment and create a relevant transaction.
```

File: saleor/product/thumbnails.py [hunk 19]
```diff
@@ -5,14 +5,13 @@
 
 @app.task
 def create_product_thumbnails(image_id):
-    """Takes a ProductImage model, and creates thumbnails for it."""
+    """Take a ProductImage model, and creates thumbnails for it."""
```
Comment (maarcingebala) [inline saleor/product/thumbnails.py]:
```suggestion
    """Take a ProductImage model and create thumbnails for it."""
```

File: saleor/product/thumbnails.py [hunk 20]
```diff
@@ -5,14 +5,13 @@
...
     create_thumbnails(pk=image_id, model=ProductImage, size_set="products")
 
 
 @app.task
 def create_category_background_image_thumbnails(category_id):
-    """Takes a Product model,
-    and creates the background image thumbnails for it."""
+    """Take a Product model, and creates the background image thumbnails for it."""
```
Comment (maarcingebala) [inline saleor/product/thumbnails.py]:
```suggestion
    """Take a Product model and create the background image thumbnails for it."""
```

File: saleor/product/thumbnails.py [hunk 21]
```diff
@@ -23,8 +22,7 @@ def create_category_background_image_thumbnails(category_id):
 
 @app.task
 def create_collection_background_image_thumbnails(collection_id):
-    """Takes a Collection model,
-    and creates the background image thumbnails for it."""
+    """Take a Collection model, and creates the background image thumbnails for it."""
```
Comment (maarcingebala) [inline saleor/product/thumbnails.py]:
```suggestion
    """Take a Collection model and create the background image thumbnails for it."""
```

**Comment by NyanKiyoshi** [comment]
@maarcingebala I applied your suggestion about Avatax, @korycins said it's great. And I fixed a mistake on a migration file (which made the CI check fail).

**Comment by maarcingebala** [comment]
Thanks @NyanKiyoshi!

**Pull request merged**